tools/rostopic/src/rostopic/__init__.py

```diff
@@ -297,7 +297,7 @@ def msgevalgen(pattern):
     for f in fields:
         if '[' in f:
             field_name, rest = f.split('[', 1)
-            if rest[-1] != ']':
+            if not rest or rest[-1] != ']':
                 print("missing closing ']' in slice spec '%s'" % f, file=sys.stderr)
```

Comment:
Exactly. That's why I didn't modified the error message. Should I add the suggested test cases?